### PR TITLE
Cart management

### PR DESCRIPTION
**Pull request by tanishk15sharma**
**CART FUNCTIONILITY**

- `Get` cart array from backend
- `Post` products in cart array to backend 
- Edit product `Qty` 
- `Delete` product from cart 
- Cart list counter on nav
- add to cart from wishlist page

### CODE REVIEW

File: src/components/nav/nav.js [hunk 1]
```diff
@@ -6,10 +6,12 @@ import { Link } from "react-router-dom";
 import { useState } from "react";
 import "./nav.css";
 import { useWishlist } from "../../contexts/wishlist-context";
+import { useCart } from "../../contexts/cart-context";
 const Nav = () => {
   const [toggleHamburger, setToggleHamburger] = useState(false);
   const { wishList } = useWishlist();
-
+  const { cart } = useCart();
+  console.log(cart);
```
Comment (Mohitkumar33) [inline src/components/nav/nav.js:14]: remove console.log

File: src/components/cartpage/CartItems.js [hunk 2]
```diff
@@ -1,16 +1,17 @@
 import "./CartItems.css";
-// import img from "../../assets/recommendedproduct-images/-Heels.webp";
```
Comment (deepak29-git) [inline src/components/cartpage/CartItems.js:2]: remove commented code

File: src/utilities/cart-utils.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      }
+    );
+    setCart(data.cart);
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+const removeFromCart = async (id, setCart) => {
+  console.log(id);
```
Comment (Mohitkumar33) [inline src/utilities/cart-utils.js:22]: remove console.log

**Review by deepak29-git — APPROVED** [review]

**Review by Mohitkumar33 — APPROVED** [review]
Nice refactoring

File: src/components/cartpage/CartItems.js (hunk 2 above)
Comment (tanishk15sharma) [reply src/components/cartpage/CartItems.js:2]: okay

File: src/components/nav/nav.js (hunk 1 above)
Comment (tanishk15sharma) [reply src/components/nav/nav.js:14]: okay , sure

**Pull request merged**